client/store/index.js

```diff
@@ -25,24 +43,71 @@ export const mutations = {
 }
 
 export const getters = {
+  authenticated (state) {
+    const hasAuthenticated = Reflect.has(state.authUser, 'authenticated')
+    let authenticated = false
+    if (hasAuthenticated) {
+      authenticated = state.authUser.authenticated
+    }
+    return authenticated
+  },
+  userTimeZone (state) {
+    const hasTimeZone = Reflect.has(state.authUser, 'tz')
+    const timeZone = 'America/New_York' // Default, in case of
+    return hasTimeZone ? state.authUser.tz : timeZone
+  },
+  userLocale (state) {
+    const hasLocale = Reflect.has(state.authUser, 'locale')
+    const locale = 'en-US' // Default, in case of
+    return hasLocale ? state.authUser.locale : locale
+  },
   authUser (state) {
     return state.authUser
   },
   isMenuHidden (state) {
     return state.isMenuHidden
+  },
+  displayName (state) {
+    const displayName = `Anonymous` // i18n? TODO
+    const hasDisplayNameProperty = Reflect.has(state.authUser, 'displayName')
+    return hasDisplayNameProperty ? state.authUser.displayName : displayName
   }
 }
 
 export const actions = {
+  /**
+   * This is run ONLY from the backend side.
+   *
+   * > If the action nuxtServerInit is defined in the store, Nuxt.js will call it with the context
+   * > (only from the server-side).
+   * > It's useful when we have some data on the server we want to give directly to the client-side.
+   *
+   * https://nuxtjs.org/guide/vuex-store#the-nuxtserverinit-action
+   * https://github.com/clarkdo/hare/blob/dev/client/store/index.js
+   * https://github.com/nuxt/docs/blob/master/en/guide/vuex-store.md
+   */
   nuxtServerInit ({ commit }, { req }) {},
-  async login ({ commit }, { userName, password, captcha }) {
+  async hydrateAuthUser ({
+    commit
+  }) {
+    const { data } = await axios.get('/hpi/auth/whois')
+    const user = Object.assign({}, data)
+    commit('SET_USER', user)
+  },
+  async login ({
+    dispatch
+  }, {
+    userName,
+    password,
+    captcha
+  }) {
     try {
-      const { data: { access_token: token } } = await axios.post('/hpi/auth/login', {
+      await axios.post('/hpi/auth/login', {
```

Comment:
Post ouf things to Koa.
If successful, it'll create a cookie payload that's a JSON string that has the JWT token in.
Since it's going to be HttpOnly, AND a Signed Cookie, we're sure the token will transit only via the cookie, and nowhere else (ideally).